src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,75 @@
-import React from 'react';
+import React, { useContext, useState } from 'react';
+import classNames from 'classnames';
+import { postComment } from '../api/comments';
+import { PostsContext } from '../PostsContext';
 
 export const NewCommentForm: React.FC = () => {
+  const [name, setName] = useState('');
+  const [isNameError, setIsNameError] = useState(false);
+  const [email, setEmail] = useState('');
+  const [isEmailError, setIsEmailError] = useState(false);
+  const [body, setBody] = useState('');
+  const [isBodyError, setIsBodyError] = useState(false);
+  const [isAddingComment, setIsAddingComment] = useState(false);
+
+  const { selectedPost, setComments } = useContext(PostsContext);
+
+  const handleNameChange = (newName: string) => {
+    setIsNameError(false);
+    setName(newName);
+  };
+
+  const handleEmailChange = (newEmail: string) => {
+    setIsEmailError(false);
+    setEmail(newEmail);
+  };
+
+  const handleBodyChange = (newBody: string) => {
+    setIsBodyError(false);
+    setBody(newBody);
+  };
+
+  const handleReset = () => {
+    setName('');
+    setIsNameError(false);
+    setEmail('');
+    setIsEmailError(false);
+    setBody('');
+    setIsBodyError(false);
+    setIsAddingComment(false);
```

Comment:
It's connected with the comment above. Think about how to optimize the reset process.